compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContext.skiko.kt

```diff
@@ -285,3 +298,141 @@ internal class DelegateRootForTestListener : PlatformContext.RootForTestListener
         }
     }
 }
+
+/**
+ * Platform specific text input service.
+ *
+ * This is a non-deprecated version of [PlatformTextInputService], which is needed because we can't
+ * expose deprecated APIs in [PlatformContext].
+ */
+interface PlatformContextTextInputService {
+    /**
+     * Start text input session for given client.
+     *
+     * @see TextInputService.startInput
+     */
+    fun startInput(
+        value: TextFieldStateAdapter,
+        imeOptions: ImeOptions,
+        onEditCommand: (List<EditCommand>) -> Unit,
+        onImeActionPerformed: ((ImeAction) -> Unit)?
+    )
+
+    /**
+     * Restart input and show the keyboard. This should only be called when starting a new
+     * `PlatformTextInputModifierNode.textInputSession`.
+     *
+     * @see TextInputService.startInput
+     */
+    fun startInput() {}
+
+    /**
+     * Stop text input session.
+     *
+     * @see TextInputService.stopInput
+     */
+    fun stopInput()
+
+    /**
+     * Request showing onscreen keyboard
+     *
+     * There is no guarantee nor callback of the result of this API.
+     *
+     * @see TextInputService.showSoftwareKeyboard
+     */
+    fun showSoftwareKeyboard()
+
+    /**
+     * Hide software keyboard
+     *
+     * @see TextInputService.hideSoftwareKeyboard
+     */
+    fun hideSoftwareKeyboard()
+
+    /**
+     * Notify the new editor model to IME.
+     *
+     * @see TextInputSession.updateState
+     */
+    fun updateState(oldValue: TextFieldStateAdapter?, newValue: TextFieldStateAdapter)
+
+    /**
+     * Notify the focused rectangle to the system.
+     *
+     * The system can ignore this information or use it to for additional functionality.
+     *
+     * For example, desktop systems show a popup near the focused input area (for some languages).
+     */
+    // TODO(b/262648050) Try to find a better API.
+    fun notifyFocusedRect(rect: Rect) {
+    }
+
+    /**
+     * Notify the input service of layout and position changes.
+     *
+     * @see TextInputSession.updateTextLayoutResult
+     */
+    fun updateTextLayoutResult(
+        textFieldValue: TextFieldStateAdapter,
+        offsetMapping: OffsetMapping,
+        textLayoutResult: TextLayoutResult,
+        textFieldToRootTransform: (Matrix) -> Unit,
+        innerTextFieldBounds: Rect,
+        decorationBoxBounds: Rect
+    ) {
+    }
+}
+
+@Suppress("DEPRECATION")
+fun PlatformContextTextInputService.asPlatformTextInputService() = object : PlatformTextInputService {
```

Comment:
If we already have `PlatformTextInputService` why do we need a new copy?

Reply:
Because `PlatformTextInputService` is deprecated.